src/emqx_acl_cache.erl

```diff
@@ -244,3 +255,12 @@ if_expired(CachedAt, Fun) ->
        true ->
            Fun(false)
     end.
+
+if_discarded() ->
+    LastEvictionStamp = erlang:get(discard_k()),
+    CurrentEvictionStamp = persistent_term:get(discard_k(), 0),
+    LastEvictionStamp =/= CurrentEvictionStamp.
```

Comment:
We should return `false | {true, NewStamp}` from this function.
And `NewStamp` should be passed to `update_discarded_stamp/1` instead of reading it again.

This is to avoid surprises when the stamp changes between these two calls.